hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/operations/FinalizeJoinOp.java

```diff
@@ -200,7 +216,16 @@ protected void readInternalImpl(ObjectDataInput in) throws IOException {
         clusterVersion = in.readObject();
         preJoinOp = readOnJoinOp(in);
         postJoinOp = readOnJoinOp(in);
+        if (deserializationFailure != null) {
+            // failure occurred while deserializing pre- or post-join ops
+            // stop deserializing now because outcome is undefined and run() will anyway fail
+            return;
+        }
         deferPartitionProcessing = in.readBoolean();
+        if (clusterVersion.isGreaterOrEqual(V5_2)) {
+            byte topologyIntentOrdinal = in.readByte();
```

Comment:
introducing an id per enum item can fit better instead of using ordinal

Reply:
done